packages/dropdown/src/Dropdown.js

```diff
@@ -1,6 +1,135 @@
+import Downshift from "downshift";
+import { TextFieldPresenter } from "@hig/text-field";
+import Icon from "@hig/icon";
+import Option from "./presenters/Option";
+import composeEventHandlers from "./composeEventHandlers";
+
 import "./dropdown.scss";
 
-Dropdown.displayName = "Dropdown";
+export default class Dropdown extends Component {
+  static propTypes = {
+    /**
+     * HTML ID attribute
+     */
+    id: PropTypes.string,
+    /**
+     * Text describing what the field represents
+     */
+    label: PropTypes.string,
+    /**
+     * Instructional text for the field
+     */
+    instructions: PropTypes.string,
+    /**
+     * Placeholder text to render when an option has not been selected
+     */
+    placeholder: PropTypes.string,
+    /**
+     * Prevents user actions on the field
+     */
+    disabled: PropTypes.bool,
+    /**
+     * Text describing why the field is required
+     */
+    required: PropTypes.string,
+    /**
+     * An array of objects to choose from.
+     */
+    options: PropTypes.arrayOf(
+      PropTypes.shape({
+        label: PropTypes.string,
+        value: PropTypes.string
+      })
+    ),
+    /**
+     * Called after user changes the value of the field. Called with the entire object that was selected.
+     */
+    onChange: PropTypes.func,
+    /**
+     * Called when user moves focus from the field
+     */
+    onBlur: PropTypes.func,
+    /**
+     * Called when user puts focus onto the field
+     */
+    onFocus: PropTypes.func
+  };
+
+  render() {
+    const {
+      id,
+      label,
+      instructions,
+      placeholder,
+      required,
+      disabled,
+      onBlur,
+      onChange,
+      onFocus,
+      options
+    } = this.props;
+
+    return (
+      <Downshift
+        onChange={onChange}
+        itemToString={item => (item ? item.label : "")}
+      >
+        {({
+          getInputProps,
+          getItemProps,
+          isOpen,
+          highlightedIndex,
+          selectedItem,
+          toggleMenu
+        }) => (
+          <div
+            className={cx("hig__dropdown", {
+              "hig__dropdown--disabled": disabled
+            })}
+          >
+            <div className="hig__dropdown__input-wrapper">
+              <TextFieldPresenter
+                {...getInputProps({
+                  id,
+                  label,
+                  instructions,
+                  required,
+                  placeholder,
+                  disabled,
+                  onBlur,
+                  onFocus: composeEventHandlers(toggleMenu, onFocus),
+                  readOnly: true // @TODO: toggle based on desired type of Dropdown
+                })}
+              />
+              <span className="hig__dropdown__input-caret">
+                {/* @TODO: there are variations of the TextField with multiple icons at the end of the input. These icon nodes should be passed as props to TextField. */}
+                <Icon name="caret" />
```

Comment:
In a refactor I didn't have time for, I would have liked TextField to be responsible for positioning any number of icon buttons at the end of an input. I spiked the interface like so:

```jsx
<TextFieldPresenter
  iconInteractions={
    <React.Fragment>
      <IconButton name="search" />
      <IconButton name="caret" />
   </React.Fragment>
  }
/>
```

For now, I absolute-position the caret to be in the correct position.